### PR TITLE
skalko.oksana

### PR DESCRIPTION
**Pull request by ok-skalko**

### CODE REVIEW

File: skalko.oksana/2homework/animal.rb [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+little_raccoon = Raccoon.new('Eli')
+p little_raccoon
+little_raccoon.eat
+little_raccoon.walk
+little_raccoon.walk
+little_raccoon.walk
+little_raccoon.walk
+little_raccoon.bathing
+little_raccoon.sleep
+little_raccoon.result
```
Comment (hawkprovvv) [inline skalko.oksana/2homework/animal.rb:80]: Please, add an [empty line](https://thoughtbot.com/blog/no-newline-at-end-of-file) at the end of all files.

File: skalko.oksana/2homework/animal.rb [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      puts "your #{@name} is too hungry"
+    end
+  end
+
+  private
+
+  def check(attribute)
+    if attribute > 100
+      100
+    elsif attribute < 0
```
Comment (hawkprovvv) [inline skalko.oksana/2homework/animal.rb]: @ok-skalko Hi) Please note rubocop comments. In this case, [prefer to use the predicate method](https://github.com/rubocop-hq/ruby-style-guide#predicate-methods) `.negative?`

File: skalko.oksana/2homework/brackets.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+      result = false
+    end
+    result
+  end
+end
+
+p "{()}".valid?
+p "(<)({)".valid?
+p "[".valid?
+p "(<>[{}])".valid?
```
Comment (hawkprovvv) [inline skalko.oksana/2homework/brackets.rb]: Please, use [single-quoted strings](https://github.com/rubocop-hq/ruby-style-guide#consistent-string-literals-single-quote) when you don't need string interpolation or special symbols.

File: skalko.oksana/2homework/brackets.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
+class String
+  def valid?
+    brack = self.chars
+    hash = {"{" => "}", "[" => "]", "(" => ")", "<" => ">"}
```
Comment (hawkprovvv) [inline skalko.oksana/2homework/brackets.rb]: Add space after the opening parenthesis and before closing parenthesis
```suggestion
    hash = { '{' => '}', '[' => ']', '(' => ')', '<' => '>' }
```

File: skalko.oksana/2homework/brackets.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    result = true
+    if brack.size.even?
+      while brack.size != 0
+        if hash[brack[0]].eql?(brack[-1])
+          brack.shift && brack.pop
+          result = true
+        elsif hash[brack[0]].eql?(brack[1])
+          brack.shift && brack.shift
```
Comment (hawkprovvv) [inline skalko.oksana/2homework/brackets.rb]: You can use `.shift(2)` instead of using identical operands

**Review by hawkprovvv — CHANGES_REQUESTED** [review]

File: skalko.oksana/2homework/animal.rb (hunk 1 above)
Comment (ok-skalko) [reply skalko.oksana/2homework/animal.rb:80]: Files in RubyMine already have empty line at the end. But github does not display them

**Review by hawkprovvv — APPROVED** [review]

**Pull request merged**